src/System.Management/src/System/Management/ManagementPath.cs

```diff
@@ -324,7 +324,7 @@ private static string GetWbemPath(IWbemPath wbemPath)
                     {
                         char[] pathChars = new char[(int)bufLen - 1];
                         status = wbemPath.GetText_(flags, ref bufLen, pathChars);
-                        pathStr = new string(pathChars);
+                        pathStr = new string(pathChars, 0, Array.IndexOf(pathChars, '\0'));;
```

Comment:
That said, if bufLen ends up being valid (I'm not familiar with the semantics of these calls), that value might be usable instead of the IndexOf call.  Basically we just need to pass in the actual length of the string (not including the null terminator).